lib/stream/cache_test.go

```diff
@@ -0,0 +1,145 @@
+package stream_test
+
+import (
+	"math/rand/v2"
+	"sync"
+	"testing"
+
+	"github.com/omni-network/omni/lib/stream"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+//go:generate go test . -run=TestCache -count=100
+
+func TestCache(t *testing.T) {
+	t.Parallel()
+
+	const limit = 100
+	cache := stream.NewMapCache[int](limit)
+
+	// Helper function ensuring length of returned elements
+	requireGetLen := func(t *testing.T, from uint64, l int) {
+		t.Helper()
+		got, err := cache.Get(from)
+		require.NoError(t, err)
+		require.Len(t, got, l)
+	}
+
+	var wg sync.WaitGroup
+	added := make(chan []int)
+
+	// Ensure it is empty
+	requireGetLen(t, 0, 0)
+	requireGetLen(t, 1, 0)
+	requireGetLen(t, limit, 0)
+
+	// Concurrently add elements to the cache
+	const writers = 10
```

Comment:
Should we add one slacking writer, so that we have a gap in the range of values?

Reply:
we do test gaps below?